staging/src/k8s.io/cli-runtime/pkg/resource/query_param_verifier.go

```diff
@@ -155,6 +156,18 @@ func supportsQueryParam(doc *openapi_v2.Document, gvk schema.GroupVersionKind, q
 			if param.GetParameter().GetNonBodyParameter().GetQueryParameterSubSchema().GetName() == string(queryParam) {
 				return true, nil
 			}
+
+			// lookup global parameters
+			if ref := param.GetJsonReference().GetXRef(); ref != "" {
+				for _, globalParam := range doc.GetParameters().GetAdditionalProperties() {
```

Comment:
this is likely to be a very long list, right? rather than iterating over the global list for every local ref we find, should we accumulate $ref names in `path.GetValue().GetPatch().GetParameters()` into a set and then iterate over the global parameter definition list once, looking for ones with the name we're looking for which are in the set referenced from `path.GetValue().GetPatch().GetParameters()`?

Reply:
done